Framework/DataHandling/src/LoadNexusLogs.cpp

```diff
@@ -646,21 +676,43 @@ void LoadNexusLogs::loadNPeriods(
  * @param workspace :: A pointer to the workspace to store the logs
  */
 void LoadNexusLogs::loadLogs(
-    ::NeXus::File &file, const std::string &entry_name,
+    ::NeXus::File &file, const std::string &absolute_entry_name,
     const std::string &entry_class,
     const boost::shared_ptr<API::MatrixWorkspace> &workspace) const {
-  file.openGroup(entry_name, entry_class);
-  std::map<std::string, std::string> entries = file.getEntries();
-  std::map<std::string, std::string>::const_iterator iend = entries.end();
-  for (std::map<std::string, std::string>::const_iterator itr = entries.begin();
-       itr != iend; ++itr) {
-    std::string log_class = itr->second;
-    if (log_class == "NXlog" || log_class == "NXpositioner") {
-      loadNXLog(file, itr->first, log_class, workspace);
-    } else if (log_class == "IXseblock") {
-      loadSELog(file, itr->first, workspace);
+
+  const std::map<std::string, std::set<std::string>> &allEntries =
+      getFileInfo()->getAllEntries();
+
+  auto lf_LoadByLogClass = [&](const std::string &logClass,
+                               const bool isNxLog) {
+    auto itLogClass = allEntries.find(logClass);
+    if (itLogClass == allEntries.end()) {
+      return;
     }
-  }
+    const std::set<std::string> &logsSet = itLogClass->second;
+    auto itPrefixBegin = logsSet.lower_bound(absolute_entry_name);
```

Comment:
Would equal_range let you avoid the string comparison in the following for loop? (I might be misreading your intent)

Reply:
That's a good thought, I tried naively using `std::upper_bound` (similar to what you point out with equal_range to get a range upfront) but I found out the hard way that for a prefix `std::lower_bound` and `std::upper_bound` return the same as any value is "larger" (from the comparison operator) than its prefix. I couldn't escape this check.

Reply:
🤕